ee/api/ee_tagged_item.py

```diff
@@ -0,0 +1,32 @@
+from django.db.models import Q
+from rest_framework import serializers
+
+from posthog.models import Tag
+
+
+class EnterpriseTaggedItemSerializerMixin(serializers.Serializer):
+    """
+    Serializer that interacts with TaggedItem model
+    """
+
+    def get_tags(self, obj):
+        return list(obj.tags.values_list("tag__name", flat=True)) if obj.tags else []
```

Comment:
I put some thoughts here about the n+1 problem here https://github.com/PostHog/posthog/pull/8529#issuecomment-1034446986. I think I'll need to rewrite the mxin to intercept the request at the viewset level rather than the serializer level in order to use `prefetch_related`.